### DISCUSSION TITLE
the best visp example/tutorial for both tracking and visual servoing (at the same time)

### DISCUSSION THREAD

**Discussion by AssanhajiMohamed** [lagadic/visp#1183] [Q&A]
Hello everyone, i am currently working on a project, where i have to manipulate a robotic arm to pick and drop some objects, i should first track them with the camera (eye-in-hand configuration) then have a visual servoing loop to command the arm, i am looking for an algorithm that hopefully can do both, or even can do the separate tasks but that we can combine.

**Comment by fspindle** [comment]
ViSP is well suited to achieve all the features you mention.

In our [tutorials](https://visp-doc.inria.fr/doxygen/visp-daily/tutorial_users.html) you will find solutions for objects detection, objects tracking and also image-based or position-based visual servoing to achieve the grasping.

**Comment by MikasaIL** [comment]
hello,have u finished this work?
